File: ecommerce-api-legacy/src/middlewares/errorHandler.js
class AppError extends Error {
  constructor(message, statusCode = 400) {
    super(message);
    this.statusCode = statusCode;
  }
}

function asyncHandler(handler) {
  return (req, res, next) => Promise.resolve(handler(req, res, next)).catch(next);
}

function notFoundHandler(req, res) {
  res.status(404).json({ error: 'Recurso não encontrado' });
}

function errorHandler(error, req, res, next) {
  const statusCode = error.statusCode || 500;
  const message = statusCode >= 500 ? 'Erro interno' : error.message;
  if (statusCode >= 500) {
    console.error(error);
  }
  res.status(statusCode).json({ error: message });
}

module.exports = { AppError, asyncHandler, errorHandler, notFoundHandler };

File: task-manager-api/middlewares/error_handler.py
from flask import jsonify


class AppError(Exception):
    def __init__(self, message, status_code=400):
        super().__init__(message)
        self.message = message
        self.status_code = status_code


def register_error_handlers(app):
    @app.errorhandler(AppError)
    def handle_app_error(error):
        return jsonify({'error': error.message}), error.status_code

    @app.errorhandler(404)
    def handle_not_found(error):
        return jsonify({'error': 'Recurso não encontrado'}), 404

    @app.errorhandler(Exception)
    def handle_unexpected_error(error):
        app.logger.exception(error)
        return jsonify({'error': 'Erro interno'}), 500
